graph_service_client.go

```diff
@@ -34,6 +34,22 @@ func NewGraphServiceClientWithCredentials(credential azcore.TokenCredential, sco
 	return client, nil
 }
 
+// NewGraphServiceClientWithCredentialsAndHosts instantiates a new GraphServiceClient with provided credentials , scopes and validhosts
+func NewGraphServiceClientWithCredentialsAndHosts(credential azcore.TokenCredential, scopes []string, validhosts []string) (*GraphServiceClient, error) {
+	auth, err := az.NewAzureIdentityAuthenticationProviderWithScopesAndValidHosts(credential, scopes, validhosts)
```

Comment:
we should set the default hosts values if nothing is passed here, this way we'll be able to clean it up in the authentication library